lib/configParser.js

```diff
@@ -58,9 +90,8 @@ var ConfigParser = function() {
  */
 var merge_ = function(into, from) {
   for (var key in from) {
-    if (into[key] instanceof Object &&
-        !(into[key] instanceof Array) &&
-        !(into[key] instanceof Function)) {
+    if ((isArrayLike_(into[key]) && isArrayLike_(from[key])) || 
+        (isDictionary_(into[key]) && isDictionary_(from[key]))) {
```

Comment:
That's a good point. But by the same token, if config has `param: {a: 2, b:3}`, and command line passes `--param.a 4`, it should override both right?
Maybe the right thing to do is convert array-like structures from command line args to arrays, and always just directly replace the value that's there in the current config?